src/rules/tictactoepoker/tictactoepoker.ts

```diff
@@ -0,0 +1,752 @@
+    }
+}
+
+export interface ScoreWithExplanations {
+    score: number,
+    explanations: ScoreExplanation[]
+}
+
+// Return score type for the given three cards (some of all of which may be empty cards)
+function compute_score(a: Card, b: Card, c: Card): ScoreType {
+    // Get all normal cards
+    let cards = [];
+    if (a.is_normal()) { cards.push(a); }
+    if (b.is_normal()) { cards.push(b); }
+    if (c.is_normal()) { cards.push(c); }
+
+    // Compute score for an array of cards, where the Aces have been resolved to
+    // either 1 or 14
+    const compute_no_ace = ((cards) => {
+
+        if (cards.length == 3) {
+
+            // Flush (all same suit)
+            const is_flush: boolean = (() => {
+                // All the same suit.
+                let suits = {};
+                for (let card of cards) {
+                    suits[card.get_suit()] = true;
+                }
+                if (Object.keys(suits).length != 1) {
+                    return false;
+                }
+
+                return true;
+            })();
+
+
+            // Straight (just three consecutive cards)
+            const is_straight: boolean = (() => {
+                if (cards[0].get_value() + 1 == cards[1].get_value() &&
+                    cards[1].get_value() + 1 == cards[2].get_value()) {
+                        return true;
+                }
+            })();
+
+            if (is_straight && is_flush) {
+                return {combination_type: CombinationType.StraightFlush, rank_card: cards[2]};
+            }
+            if (is_flush) {
+                // Find the card with the highest rank value - it can be anywhere in the three.
+                let best_rank_value = 0;
+                let best_card = Card.EMPTY;
+                for (let card of cards) {
+                    if (card.get_rank_value() > best_rank_value) {
+                        best_rank_value = card.get_rank_value();
+                        best_card = card;
+                    }
+                }
+                return {combination_type: CombinationType.Flush, rank_card: best_card};
+            }
+            if (is_straight) {
+                return {combination_type: CombinationType.Straight, rank_card: cards[2]};
+            }
+
+            // Three of a kind
+            if (cards[0].get_value() == cards[1].get_value() &&
+                cards[1].get_value() == cards[2].get_value()) {
+                return {combination_type: CombinationType.ThreeOfAKind, rank_card: cards[0]};
+            }
+        }
+
+        // Pair
+        const pair_card: Card = (() => {
+            for (let i = 0; i < cards.length; i++) {
+                for (let j = i + 1; j < cards.length; j++) {
+                    if (cards[i].get_value() == cards[j].get_value()) {
+                        return cards[i];
+                    }
+                }
+            }
+            return Card.EMPTY;
+        })();
+
+        if (pair_card != Card.EMPTY) {
```

Comment:
same thing - !==

Reply:
ESLint doesn't complain. What is being implicitly casted?